src/hooks.ts

```diff
@@ -48,15 +48,38 @@ function useCallsDataSubscription(
     }
   }, [actions, chainId, dispatch, blocksPerFetch, serializedCallKeys])
 
-  return useMemo(
-    () =>
-      calls.map<CallResult>((call) => {
-        if (!chainId || !call) return INVALID_RESULT
-        const result = callResults[chainId]?.[toCallKey(call)]
-        const data = result?.data && result.data !== '0x' ? result.data : undefined
-        return { valid: true, data, blockNumber: result?.blockNumber }
-      }),
-    [callResults, calls, chainId]
+  // ensures that call results arrays remain referentially equivalent when unchanged to prevent
+  // spurious re-renders, which would otherwise occur because mapping always creates a new object
+  return useStabilizeCallResults(
+    useMemo(
+      () =>
+        calls.map<CallResult>((call) => {
+          if (!chainId || !call) return INVALID_RESULT
+          const result = callResults[chainId]?.[toCallKey(call)]
+          const data = result?.data && result.data !== '0x' ? result.data : undefined
+          return { valid: true, data, blockNumber: result?.blockNumber }
+        }),
+      [callResults, calls, chainId]
+    )
+  )
+}
+
+function useStabilizeCallResults(results: CallResult[]): CallResult[] {
```

Comment:
i think theres a better way to do this, eg where we are producing the callresult object

using state feels like it would have a lot of overhead